android/KMEA/app/src/main/java/com/tavultesoft/kmea/LanguageSettingsActivity.java

```diff
@@ -58,17 +60,26 @@ public void onCreate(Bundle savedInstanceState) {
     getSupportActionBar().setDisplayShowHomeEnabled(true);
     getSupportActionBar().setDisplayShowTitleEnabled(false);
     TextView textView = (TextView) findViewById(R.id.bar_title);
-    textView.setText(getString(R.string.title_language_settings));
 
     listView = (ListView) findViewById(R.id.listView);
     listView.setFastScrollEnabled(true);
 
     Bundle bundle = getIntent().getExtras();
+    FilteredKeyboardsAdapter adapter = null;
     if (bundle != null) {
```

Comment:
Can we negate this test so we can avoid ever having `adapter = null` in the code flow, i.e.
```
  if(bundle == null) {
      // Should never actually happen.
      Log.v(TAG, "Language data not specified for LanguageSettingsActivity!");
      finish();
      return; //??is this needed
  }
  ...
  FilteredKeyboardsAdapter adapter = new FilteredKeyboardsAdapter...
```